### PR TITLE
feat: updated configuration vars to be consistent with other agents

### PR DESCRIPTION
**Pull request by shaleynikov**

### CODE REVIEW

**Comment by shaleynikov** [comment]
Depends on #13

File: src/index.ts [hunk 1]
```diff
@@ -9,42 +9,45 @@ type TagList = Record<string, any>
...
+    appName: 'nodejs',
     tags: {},
   }
 ): void {
   if (c) {
-    config.server = c.server || DEFAULT_SERVER
+    config.serverAddress = c.serverAddress || DEFAULT_SERVER
     config.sourceMapPath = c.sourceMapPath
-    config.name = c.name || 'nodejs'
+    config.appName = c.appName || 'nodejs'
```
Comment (eh-am) [inline src/index.ts:49]: Based on the go implementation, `applicationName` is required https://github.com/pyroscope-io/client/blob/563bf89a859604a1cca9bc01503e8b4de50b6f05/internal/flameql/error.go#L15

I guess we should do the same?

File: src/index.ts [hunk 2]
```diff
@@ -9,42 +9,45 @@ type TagList = Record<string, any>
...
 
-
 export function init(
   c: Partial<PyroscopeConfig> = {
-    server: DEFAULT_SERVER,
-    autoStart: true,
-    name: 'nodejs',
+    serverAddress: DEFAULT_SERVER,
+    appName: 'nodejs',
     tags: {},
```
[the remark this answers is not in the document]
Comment (eh-am) [reply src/index.ts:43]: Hmm there are defaults also defined in line 31? I think that's a bit confusing, can we simplify all this?

**Review by eh-am — COMMENTED** [review]
Admittedly I just looked at go, but from what I saw the env var integration is done at the application level (ie the application expects an `appName`, how the application figures that out via envvar/config file is up to the application), so I don't quite get why we are doing it here.

File: src/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (shaleynikov) [reply src/index.ts]: yeah, simplifed

File: src/index.ts (hunk 1 above)
Comment (shaleynikov) [reply src/index.ts]: Made app name mandatory

**Comment by shaleynikov** [comment]
> Admittedly I just looked at go, but from what I saw the env var integration is done at the application level (ie the application expects an `appName`, how the application figures that out via envvar/config file is up to the application), so I don't quite get why we are doing it here.

Well, other agrents, does it differently. Like java is completely configured via env vars. User is still able to configure on app level, just left a possibility to use env vars. I can delete if it isn't consistent

**Review by petethepig — APPROVED** [review]
Looks good. Let's release it!

**Pull request merged**